Add support for index pattern exclusion in CCR AutoFollow (#72935)
elastic/elasticsearch: docs/reference/ccr/apis/auto-follow/put-auto-follow-pattern.asciidoc

  [
    "<leader_index_pattern>"
  ],
  "leader_index_exclusion_patterns":
  [

>     "leader_index_exclusion_pattern>"

Reviewer: ```suggestion
    "<leader_index_exclusion_pattern>"
```

  (Optional, array) An array of simple index patterns to match against indices
  in the remote cluster specified by the `remote_cluster` field.

`leader_index_exclusion_patterns`::

>   (Optional, array) An array of simple index patterns to avoid following
>   indices in the remote cluster specified by the `remote_cluster` field.

Reviewer: ```suggestion
  (Optional, array) An array of simple index patterns that will not be
  auto-followed.
```

>   (Optional, array) An array of simple index patterns to avoid following
>   indices in the remote cluster specified by the `remote_cluster` field.

Reviewer 2: Maybe?

```suggestion
  (Optional, array) An array of simple index patterns that can be used to exclude indices from being auto-followed. Indices in the remote cluster whose names are matching one or more `leader_index_patterns` and one or more `leader_index_exclusion_patterns` won't be followed.
```
